File: leet_704_binary_search__Iter.cpp
class Solution {
public:
    int search(vector<int>& nums, int target)
    {
        int n = nums.size() ;
        int R = n - 1;
        int L = 0;
        while (L <= R)
        {
            int mp = (L + R) / 2;
            if (nums[mp] == target)
                return mp;
            if (nums[mp] > target)
                R = mp - 1;
            else
                L = mp + 1;
        }
        return -1;
    }
};

File: leet_704_binary_search__Iter.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */

var search = function(arr, target) {
    n = arr.length
    if (n == 1){
        if (arr[0] == target)
            return 0
        else
            return -1
    }
    R = n - 1
    L = 0
    while (L <= R){
        mid = Math.floor((L + R) / 2)
        if (arr[mid] == target)
            return mid
        else if (arr[mid] > target)
            R = mid - 1
        else
            L = mid + 1
    }
    return -1
};
